### ISSUE TITLE
MAINT - Replace flake8 with ruff

### ISSUE THREAD

**Issue by trallard** [nebari-dev/nebari#1589]
It was recently brought up that flake8 could be replaced with [Ruff](https://github.com/charliermarsh/ruff) and other projects in the ecosystem have started migrating to it.

See original post: https://github.com/nebari-dev/nebari/pull/1545#issuecomment-1326021429

**Comment by costrouc** [comment]
Makes sense. I did see it has a drop in replacement for flake8 settings as well and no need for us to adopt the other format options for now in ruff.
